--------------------------------------------------------------------------------------------------------------------------------------------
## Background:

Yersinia pestis is infamous for causing plague, such as the Bubonic plague and the Justinian plague, killing millions.

Antiqua strain is thought to be the ancestral branch of the plague pathogen (https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1482938/).

Specifically, the pMT1 plasmid found in Yersinia Pestis is a virulence factor that promotes tissue invasion (https://www.ncbi.nlm.nih.gov/pmc/articles/PMC108724/). 

Several whole genome sequnces of the pMT1 plasmid from the bacteria were gathered from GenBank and NCBI's RefSeq.

Sequences collected were comprised of various strains of pMT1. 

Searches of the strains on Joint Genome Institute produced scientific study citations that mentioned the origin of some strains.

Confirmed by accession number, using NCBI assembly to match RefSeq and Genbank numbers. Strains found to be having unknown origins or made in a lab or were removed.


**Raw Sequence Information**
Yersinia pestis Antiqua plasmid pMT, complete sequence
GenBank: CP000309.1
Info: Isolated from a human infection in Africa (Republic of Congo in 1965) (https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1482938/, https://img.jgi.doe.gov/cgi-bin/m/main.cgi?section=TaxonDetail&page=taxonDetail&taxon_oid=637000350)
https://www.ncbi.nlm.nih.gov/nuccore/CP000309.1?report=fasta

Yersinia pestis Nepal516 plasmid pMT, complete sequence
GenBank: CP000306.1
Info: Isolated from a human infection in Nepal (possibly from a 1967 outbreak of pneumonic plague) (https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1482938/, genbank=refseq: https://www.ncbi.nlm.nih.gov/assembly/GCA_000013805.1, https://img.jgi.doe.gov/cgi-bin/m/main.cgi?section=TaxonDetail&page=taxonDetail&taxon_oid=645058759)
https://www.ncbi.nlm.nih.gov/nuccore/CP000306.1?report=fasta

Yersinia pestis biovar Microtus str. 91001 plasmid pMT1, complete sequence
GenBank: AE017045.1
Info: Isolated from Microtus brandti in Inner Mongolia, China (https://pubmed.ncbi.nlm.nih.gov/15368893/, https://img.jgi.doe.gov/cgi-bin/m/main.cgi?section=TaxonDetail&page=taxonDetail&taxon_oid=637000354) 
https://www.ncbi.nlm.nih.gov/nuccore/AE017045.1?report=fasta

Yersinia pestis biovar Medievalis str. Harbin 35 plasmid pMT, complete sequence
NCBI Reference Sequence: NC_017266.1
Info: Isolated from a human in China (https://www.ncbi.nlm.nih.gov/bioproject/?term=PRJNA30505, refseq=genbank: https://www.ncbi.nlm.nih.gov/assembly/GCF_000186725.1/, https://img.jgi.doe.gov/cgi-bin/m/main.cgi?section=TaxonDetail&page=taxonDetail&taxon_oid=650377987)
https://www.ncbi.nlm.nih.gov/nuccore/NC_017266.1?report=fasta

Yersinia pestis strain Cadman plasmid pMT1, complete sequence
NCBI Reference Sequence: NZ_CP016275.1
Info: Isolated in US 1965 from a boy's CSF (https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5084870/, https://img.jgi.doe.gov/cgi-bin/m/main.cgi?section=TaxonDetail&page=taxonDetail&taxon_oid=2718217770)
https://www.ncbi.nlm.nih.gov/nuccore/NZ_CP016275.1?report=fasta

Yersinia pestis strain S19960127 plasmid pMT1, complete sequence
NCBI Reference Sequence: NZ_CP045637.1
Info: Isolated from a pneumonic patient’s organs at necropsy during a plague outbreak that occurred in 1996 in Qayü village, Lhünze County, Shannan Prefecture in Tibet, China (https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8096067/, )
https://www.ncbi.nlm.nih.gov/nuccore/NZ_CP045637.1?report=fasta

Yersinia pestis CA88-4125 plasmid pMT1, complete sequence
NCBI Reference Sequence: NC_009596.1
Info: Isolated from human 1988 California (refseq=genbank: https://www.ncbi.nlm.nih.gov/assembly/GCF_000181455.2/, https://img.jgi.doe.gov/cgi-bin/m/main.cgi?section=TaxonDetail&page=taxonDetail&taxon_oid=640963031)
https://www.ncbi.nlm.nih.gov/nuccore/NC_009596.1?report=fasta

Yersinia pestis CO92 plasmid pMT1, complete sequence
NCBI Reference Sequence: NC_003134.1
Info: Isolated from a US human who got it from a cat at or before 2001 (https://core.ac.uk/reader/13110408?utm_source=linkout, genbank=refseq: https://www.ncbi.nlm.nih.gov/assembly/GCF_000009065.1/, https://img.jgi.doe.gov/cgi-bin/m/main.cgi?section=TaxonDetail&page=taxonDetail&taxon_oid=637000351)
https://www.ncbi.nlm.nih.gov/nuccore/NC_003134.1?report=fasta

Yersinia pestis D106004 plasmid pMT1, complete sequence
NCBI Reference Sequence: NC_017155.1
Info: Isolated from Apodemus chevrieri in Yulong county in 2006 (https://www.ajtmh.org/view/journals/tpmd/81/4/article-p714.xml, refseq=genbank: https://www.ncbi.nlm.nih.gov/assembly/GCF_000022805.1/, https://img.jgi.doe.gov/cgi-bin/m/main.cgi?section=TaxonDetail&page=taxonDetail&taxon_oid=646862350)
https://www.ncbi.nlm.nih.gov/nuccore/NC_017155.1?report=fasta

Yersinia pestis Z176003 plasmid pMT1, complete sequence
NCBI Reference Sequence: NC_014022.1
Info: Isolated from dead marmot in Tibet Autonomous Region, China (refseq=genbank: https://www.ncbi.nlm.nih.gov/assembly/GCF_000022845.1/, https://img.jgi.doe.gov/cgi-bin/m/main.cgi?section=TaxonDetail&page=taxonDetail&taxon_oid=646564590)
https://www.ncbi.nlm.nih.gov/nuccore/NC_014022.1?report=fasta

Yersinia pestis Pestoides G plasmid pMT sequence
GenBank: CP010248.1
Info: Isolated from Microtus montanus in Republic of Georgia (https://img.jgi.doe.gov/cgi-bin/m/main.cgi?section=TaxonDetail&page=taxonDetail&taxon_oid=2648501335)
https://www.ncbi.nlm.nih.gov/nuccore/CP010248.1?report=fasta

Yersinia pestis subsp. pestis bv. Medievalis strain SCPM-O-B-6530 plasmid pMT, complete sequence
GenBank: CP045159.1
Info: Isolated from Citellophilus tesquorum fleas in Central-Caucasian high-mountain, Russia 2000 (https://journals.asm.org/doi/10.1128/mra.01115-21, https://img.jgi.doe.gov/cgi-bin/m/main.cgi?section=TaxonDetail&page=taxonDetail&taxon_oid=2891959310)
https://www.ncbi.nlm.nih.gov/nuccore/CP045159.1?report=fasta

Yersinia pestis Pestoides F plasmid pMT, complete sequence
GenBank: CP009714.1
Info: Isolated from host in former Soviet Union (https://link.springer.com/chapter/10.1007/978-0-387-72124-8_2, https://img.jgi.doe.gov/cgi-bin/m/main.cgi?section=TaxonDetail&page=taxonDetail&taxon_oid=2630968993)
https://www.ncbi.nlm.nih.gov/nuccore/CP009714.1?report=fasta

--------------------------------------------------------------------------------------------------------------------------------------------
## Goal:
**Understand the evolutionary history of Yersinia pestis based on pMT plasmid evolution**

--------------------------------------------------------------------------------------------------------------------------------------------
## Step 0: Create an organized directory

**Completed - Organizing folders**
- Created a "phylogenetics-final" folder. Inside of the "phylogenetics-final" folder, created (1) "notebook-log.md" file, which explained all steps taken; (2) "data" folder, which
contained all collected data, softwares, and input and output files; (3) "README.md" file, which provided a quick summary of the project.

## Step 1: Collect data

**Completed - Collecting data**
- Data received from NCBI's GenBank and stored separately in the "raw-data" subfolder found within the "data" folder. Data chosen based on being around 100,000 nucleotides
long, which matches the average sequence of the pMT1 genome. Chose only complete sequences of pMT1.



## Step 2: Quality control

**Completed - Quality control the data**
- Data received in FASTA form so already quality check was already completed and confirmed by NCBI.



## Step 3: Alignment

**Completed - using ClustalW**

- Description: ClustalW is a software used for multiple sequence alignment of nucleotides. It is the most user friendly and performs quality alignments using progressive alignment methods.

- Assumptions: all aligned sequences are evolutionarily related, rate of change in sequences is constant

- Limitations: struggles to align distantly related sequences, takes very long to align big number of sequences

_Code:_

```
#Downloaded clustalw in the "data" folder. Entered the "data" folder via terminal. Created a fasta file containing all sequences called "pestis-pMT1-all.fasta" to be the input file. Ran alignment with the following commands to get an output file called "pestis-pMT1-all-clustalw.fasta".

clustalw2 -ALIGN -INFILE=pestis-pMT1-all.fasta -OUTFILE=pestis-pMT1-all-clustalw.fasta -OUTPUT=FASTA

```


**Completed - using Mafft**

- Description: Mafft is a multiple sequence alignment tool that progressively aligns large sequences quickly and is macos friendly.

- Assumptions: input sequences are homologous (from a common ancestor) (https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3603318/), all lineages have the same evolutionary rate (https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4920119/)

- Limitations: has no function to exclude divergent input sequences (https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4920119/)

_Code:_

```
#Mafft was already downloaded in the "data" folder. Entered the "data" folder via terminal. Used previously created input fasta file containing all sequences called "pestis-pMT1-all.fasta". Ran alignment with the following commands to get an output file called "pestis-pMT1-all-mafft.fasta".

mafft pestis-pMT1-all.fasta > pestis-pMT1-all-mafft.fasta

```

## Step 4: TrimAl

**Completed - using Trimal**

- Description: 

- Assumptions:

- Limitations:

_Code:_

```
#mafft-aligned Trimal trimming

#Downloaded Trimal in the "data" folder. Entered the "data" folder via terminal. Ran Trimal using the previously created input file "pestis-pMT1-all-mafft.fasta" to get an output file called "pestis-pMT1-all-mafft-trimal.fasta".

trimal -in pestis-pMT1-all-mafft.fasta -out pestis-pMT1-all-mafft-trimal.fasta

```

```
#clustalw-aligned Trimal trimming

#Trimal was already downloaded in the "data" folder. Entered the "data" folder via terminal. Ran Trimal using the previously created input file "pestis-pMT1-all-clustalw.fasta" to get an output file called "pestis-pMT1-all-clustalw-trimal.fasta".

trimal -in pestis-pMT1-all-clustalw.fasta -out pestis-pMT1-all-clustalw-trimal.fasta

```



## Step 5: Distance-based tree and parsimony-based tree using the ape and phangorn R packages

**Completed- Parsimony Tree per each Clustalw/Trimal and Mafft/Trimal Alignment with selecting CP000309.1 as root**

- Description: The ape and phangorn R packages will create distance-based trees and parsimony-based trees. Parsimony-based methods will create a tree that minimizes the amount of evolutionary change required to explain the data.

- Assumptions:  Parsimony methods assume independence among characters. 

- Limitations: Parsimony-based may not produce an accurate tree due to assuming the simplest explanation of evolutionary change is correct; trees may be inconsistent.

- Note of how to root trees: root all trees using root(tre.ini,outgroup=“name of clade plasmids”) or root(tre.ini, node=number)

_Code:_

```

#mafft-aligned parsimony tree:

#Opened software R and inputted the following commands to obtain a tree, which was saved to the desktop under "pMT-mafft-trimal-parsimony.pdf"

library(ape)

library(adegenet)

library(phangorn)

dna <- fasta2DNAbin(file="/Users/rajvipatel/phylogenetics-final/data/pestis-pMT1-all-mafft-trimal.fasta")

dna2 <- as.phyDat(dna)

tre.ini <- nj(dist.dna(dna,model="raw"))

parsimony(tre.ini, dna2)

tre.pars <- optim.parsimony(tre.ini, dna2)

plot(tre.ini)

nodelabels()

tre2 = tre2 <- root(tre.ini, outgroup = "CP000309.1 263294 bp")

plot(tre2, cex = 0.6)

title("pMT mafft-aligned parsimony-based tree")

```

```

#clustalw-aligned parsimony tree:

#Opened software R and inputted the following commands to obtain a tree, which was saved to the desktop under "pMT-clustalw-trimal-parsimony.pdf"

library(ape)

library(adegenet)

library(phangorn)

dna <- fasta2DNAbin(file="/Users/rajvipatel/phylogenetics-final/data/pestis-pMT1-all-clustalw-trimal.fasta")

dna2 <- as.phyDat(dna)

tre.ini <- nj(dist.dna(dna,model="raw"))

parsimony(tre.ini, dna2)

tre.pars <- optim.parsimony(tre.ini, dna2)

plot(tre.ini)

nodelabels()

tre2 = root(tre.ini,outgroup="CP000309.1 210104 bp")

plot(tre2, cex = 0.6)

title("pMT clustalw-aligned parsimony-based tree")

```

**Completed- Distance Tree per each Clustalw/Trimal and Mafft/Trimal Alignment with selecting CP000309.1 as root**

- Description: The ape and phangorn R packages will create distance-based trees and parsimony-based trees. Distance-based methods will create a tree by calculating pairwise distances between molecular sequences. 

- Assumptions:  Distance methods assume rate of molecular evolution is constant over time and across lineages.

- Limitations: Distance-based relies on an existing model and which model is chosen; there is no flexibility.

- Note of how to root trees: root all trees using root(tre.ini,outgroup=“name of clade plasmids”) or root(tre.ini, node=number)

_Code:_

```

#mafft-aligned distance tree:

#Opened software R and inputted the following commands to obtain a tree, which was saved to the desktop under "pMT-mafft-trimal-distance.pdf"

library(ape)

library(adegenet)

library(phangorn)

dna <- fasta2DNAbin(file="/Users/rajvipatel/phylogenetics-final/data/pestis-pMT1-all-mafft-trimal.fasta")

D <- dist.dna(dna, model="TN93")

tre <- nj(D)

plot(tre)

nodelabels()

tre2 = root(tre,outgroup="CP000309.1 263294 bp")

tre <- ladderize(tre)

plot(tre, cex=.6)
title("pMT mafft-aligned distance-based tree")

```

```

#clustalw-aligned distance tree:

#Opened software R and inputted the following commands to obtain a tree, which was saved to the desktop under "pMT-clustalw-trimal-distance.pdf"

library(ape)

library(adegenet)

library(phangorn)

dna <- fasta2DNAbin(file="/Users/rajvipatel/phylogenetics-final/data/pestis-pMT1-all-clustalw-trimal.fasta")

D <- dist.dna(dna, model="TN93")

tre <- nj(D)

plot(tre)

nodelabels()

tre2 = root(tre,outgroup="CP000309.1 210104 bp")

tre <- ladderize(tre)

plot(tre, cex=.6)
title("pMT clustalw-aligned distance-based tree")

```


## Step 6: Maximum Likelihood Inference Method Per each Parsimony and Distance Tree

**Completed- RAxML-NG Maximum Likelihood Tree per each Clustalw/Trimal and Mafft/Trimal Alignment with selecting CP000309.1 as root**

- Description: RAxML-NG is a maximum likelihood tool that infers phylogenetic trees using any of the following input sequence options: nucleotides, amino acids, binary/multi-state morphological data, RNA secondary structure.

- Assumptions: Mutation process is the same at every branch of the tree. All sites evolve independently of each other.  All sites evolve the same.

- Limitations: Has higher variance of trees

_Code:_

```
#Example:

Rajvis-Air:raxml-ng_v1.1.0_macos_x86_64 rajvipatel$ ./raxml-ng --check --msa ng-tutorial/bad.fa --model GTR+G
Rajvis-Air:raxml-ng_v1.1.0_macos_x86_64 rajvipatel$ ./raxml-ng --check --msa ng-tutorial/bad.fa.raxml.reduced.phy --model GTR+G
Rajvis-Air:raxml-ng_v1.1.0_macos_x86_64 rajvipatel$ ./raxml-ng --parse --msa ng-tutorial/prim.phy --model GTR+G
Rajvis-Air:raxml-ng_v1.1.0_macos_x86_64 rajvipatel$ ./raxml-ng --msa ng-tutorial/prim.phy --model GTR+G --prefix T3 --threads 2 --seed 2

```

```

#mafft-aligned raxml-ng tree:

#Downloaded raxml-ng into the "data" folder. Using macos "Finder" app, entered the "phylogenetics-final folder" then went to the "data" folder. Duplicated the "pestis-pMT1-all-mafft-trimal.fasta" file and renamed it "pestis-pMT1-all-mafft-trimal-raxml.fasta". Dragged the "pestis-pMT1-all-mafft-trimal-raxml.fasta" file into the "raxml-ng_v1.1.0_macos_x86_64" folder. Entered the "raxml-ng_v1.1.0_macos_x86_64" folder on terminal. Ran the following commands on the "pestis-pMT1-all-mafft-trimal-raxml.fasta" input file.

./raxml-ng --check --msa pestis-pMT1-all-mafft-trimal-raxml.fasta --model GTR+G

./raxml-ng --check --msa pestis-pMT1-all-mafft-trimal-raxml.fasta.raxml.reduced.phy --model GTR+G

./raxml-ng --parse --msa pestis-pMT1-all-mafft-trimal-raxml.fasta.raxml.reduced.phy --model GTR+G

./raxml-ng --msa pestis-pMT1-all-mafft-trimal-raxml.fasta.raxml.reduced.phy --model GTR+G --prefix T5 --threads 2 --search autoMRE=0.01

#Using macos "Finder" app, found the "phylogenetics-folder" with the objective of entering the "raxml-ng_v1.1.0_macos_x86_64" folder. Inside of the "raxml-ng_v1.1.0_macos_x86_64" folder, opened the "T5.raxml.bestTree" file. Copied the Newick formatted data to use in the following commands on software R. Opened software R and inputted the following commands to obtain a tree, which was saved to the desktop under "pMT-mafft-trimal-raxml.pdf"

library(ape)

library(adegenet)

library(phangorn)

mytree <- read.tree(text='((((CP010248.1_263294_bp:0.000043,CP009714.1_263294_bp:0.000030):0.000124,((((((AE017045.1_263294_bp:0.000135,NC_017266.1_263294_bp:0.000168):0.003031,(CP045159.1_263294_bp:0.000142,NC_003134.1_263294_bp:0.000055):0.014568):0.195463,NC_017155.1_263294_bp:0.000067):0.000012,CP000306.1_263294_bp:0.000068):0.000017,NC_014022.1_263294_bp:0.000001):0.000001,CP000309.1_263294_bp:0.000054):0.000107):0.324203,NZ_CP045637.1_263294_bp:0.006637):0.000001,NC_009596.1_263294_bp:0.000001,NZ_CP016275.1_263294_bp:0.000001);')

plot(mytree)

nodelabels()

tre2 = root(mytree,outgroup="CP000309.1_263294_bp")

plot(tre2, cex = 0.6)
title("pMT mafft-aligned raxml maximum likelihood-based tree")

```

```

#clustalw-aligned raxml-ng tree:

#Raxml-ng was already downloaded into the "data" folder. Using macos "Finder" app, entered the "phylogenetics-final folder" then went to the "data" folder. Duplicated the "pestis-pMT1-all-clustalw-trimal.fasta" file and renamed it "pestis-pMT1-all-clustalw-trimal-raxml.fasta". Dragged the "pestis-pMT1-all-clustalw-trimal-raxml.fasta" file into the "raxml-ng_v1.1.0_macos_x86_64" folder. Entered the "raxml-ng_v1.1.0_macos_x86_64" folder on terminal. Ran the following commands on the "pestis-pMT1-all-clustalw-trimal-raxml.fasta" input file.

./raxml-ng --check --msa pestis-pMT1-all-clustalw-trimal-raxml.fasta --model GTR+G

./raxml-ng --check --msa pestis-pMT1-all-clustalw-trimal-raxml.fasta.raxml.reduced.phy --model GTR+G

./raxml-ng --parse --msa pestis-pMT1-all-clustalw-trimal-raxml.fasta.raxml.reduced.phy --model GTR+G

./raxml-ng --msa pestis-pMT1-all-clustalw-trimal-raxml.fasta.raxml.reduced.phy --model GTR+G --prefix T6 --threads 2 --search autoMRE=0.01)(without seed so it runs until likelihood score between 2 iterations is less than 1% of current likelihood score)

#Using macos "Finder" app, found the "phylogenetics-folder" with the objective of entering the "raxml-ng_v1.1.0_macos_x86_64" folder. Inside of the "raxml-ng_v1.1.0_macos_x86_64" folder, opened the "T6.raxml.bestTree" file. Copied the Newick formatted data to use in the following commands on software R. Opened software R and inputted the following commands to obtain a tree, which was saved to the desktop under "pMT-clustalw-trimal-raxml.pdf"

library(ape)

library(adegenet)

library(phangorn)

mytree <- read.tree(text='(((NC_009596.1_210104_bp:0.000001,NZ_CP016275.1_210104_bp:0.000001):0.000001,NZ_CP045637.1_210104_bp:0.018248):100.000000,(CP000306.1_210104_bp:0.000085,NC_017155.1_210104_bp:0.000095):0.000008,((((CP009714.1_210104_bp:0.000033,CP010248.1_210104_bp:0.000034):0.221188,(((CP045159.1_210104_bp:0.000399,NC_003134.1_210104_bp:0.000343):0.832370,NC_017266.1_210104_bp:0.000153):0.000001,AE017045.1_210104_bp:0.000126):0.198481):0.215744,CP000309.1_210104_bp:0.000067):0.000001,NC_014022.1_210104_bp:0.000001):0.000006);')

plot(mytree)

nodelabels()

tre2 = root(mytree,outgroup="CP000309.1_210104_bp")

plot(tre2, cex = 0.6)
title("pMT clustalw-aligned raxml maximum likelihood-based tree")

```

**Completed- IQ-Tree 2 Maximum Likelihood Tree per each Clustalw/Trimal and Mafft/Trimal Alignment with selecting CP000309.1 as root**

- Description: IQ-Tree 2 is a program that creates maximum likelihood phylogenies. The three main steps include using a  fast model selection via ModelFinder, an effective tree search algorithm, and an ultrafast bootstrap approximation.

- Assumptions: samples local optima so top optima is the ideal tree, 

- Limitations: longer running time, 100 maximum perturbations, no guarantee that output phylogeny represents global optimum

_Code:_

```

#Example:

Rajvis-MacBook-Air:iqtree-2.2.0-MacOSX rajvipatel$ bin/iqtree2 -s input.fasta


```

```

#mafft-aligned iqtree2 tree:

#Downloaded iqtree2 into the "data" folder. Using macos "Finder" app, entered the "phylogenetics-final folder" then went to the "data" folder. Duplicated the "pestis-pMT1-all-mafft-trimal.fasta" file and renamed it "pestis-pMT1-all-mafft-trimal-iqtree.fasta". Dragged the "pestis-pMT1-all-mafft-trimal-iqtree.fasta" file into the "iqtree-2.2.0-MacOSX" folder. Entered the "iqtree-2.2.0-MacOSX" folder on terminal. Ran the following commands on the "pestis-pMT1-all-mafft-trimal-iqtree.fasta" input file.

bin/iqtree2 -s pestis-pMT1-all-mafft-trimal-iqtree.fasta

#Using macos "Finder" app, found the "phylogenetics-folder" with the objective of entering the "iqtree-2.2.0-MacOSX" folder. Inside of the "iqtree-2.2.0-MacOSX" folder, opened the pestis-pMT1-all-mafft-trimal-iqtree.fasta.treefile file. Copied the Newick formatted data to use in the following commands on software R. Opened software R and inputted the following commands to obtain a tree, which was saved to the desktop under "pMT-mafft-trimal-iqtree.pdf"

library(ape)

library(adegenet)

library(phangorn)

mytree <- read.tree(text='(CP000309.1:0.0000541585,((CP000306.1:0.0000676246,(((AE017045.1:0.0001346761,NC_017266.1:0.0001681082):0.0030345631,(NC_003134.1:0.0000547773,CP045159.1:0.0001426097):0.0145850616):0.1952032729,NC_017155.1:0.0000670555):0.0000121843):0.0000172803,NC_014022.1:0.0000003798):0.0000003798,(((NZ_CP016275.1:0.0000007861,NC_009596.1:0.0000003798):0.0000008566,NZ_CP045637.1:0.0066396013):0.3234548284,(CP010248.1:0.0000429104,CP009714.1:0.0000296895):0.0001242404):0.0001075467);')

plot(mytree)

nodelabels()

tre2 = root(mytree,outgroup="CP000309.1")

plot(tre2, cex = 0.6)
title("pMT mafft-aligned iqtree maximum likelihood-based tree")

```

```

#clustalw-aligned iqtree2 tree:

#Iqtree2 was already downloaded into the "data" folder. Using macos "Finder" app, entered the "phylogenetics-final folder" then went to the "data" folder. Duplicated the "pestis-pMT1-all-clustalw-trimal.fasta" file and renamed it "pestis-pMT1-all-clustalw-trimal-iqtree.fasta". Dragged the "pestis-pMT1-all-clustalw-trimal-iqtree.fasta" file into the "iqtree-2.2.0-MacOSX" folder. Entered the "iqtree-2.2.0-MacOSX" folder on terminal. Ran the following commands on the "pestis-pMT1-all-clustalw-trimal-iqtree.fasta" input file.

bin/iqtree2 -s pestis-pMT1-all-clustalw-trimal-iqtree.fasta

#Using macos "Finder" app, found the "phylogenetics-folder" with the objective of entering the "iqtree-2.2.0-MacOSX" folder. Inside of the "iqtree-2.2.0-MacOSX" folder, opened the pestis-pMT1-all-clustalw-trimal-iqtree.fasta.treefile file. Copied the Newick formatted data to use in the following commands on software R. Opened software R and inputted the following commands to obtain a tree, which was saved to the desktop under "pMT-clustalw-trimal-iqtree.pdf"

library(ape)

library(adegenet)

library(phangorn)

mytree <- read.tree(text='(AE017045.1:0.0001455069,NC_017266.1:0.0001286336,((NC_003134.1:0.0004308973,CP045159.1:0.0005177772):1.2539311755,((((CP000309.1:0.0000669229,NC_014022.1:0.0000004754):0.0000010413,(CP000306.1:0.0000830916,NC_017155.1:0.0000933742):0.0000128895):0.0312880351,((NZ_CP016275.1:0.0000009620,NC_009596.1:0.0000004754):0.0000009591,NZ_CP045637.1:0.0189830752):6.4242885658):0.2307907686,(CP010248.1:0.0000203969,CP009714.1:0.0000452803):0.2998268146):0.2222226247):0.0292935461);')

plot(mytree)

nodelabels()

tre2 = root(mytree,outgroup="CP000309.1")

plot(tre2, cex = 0.6)
title("pMT clustalw-aligned iqtree maximum likelihood-based tree")

```

## Step 7: Bayesian Method Per each Parsimony and Distance Tree

**Not completed- Mr.Bayes per each Clustalw/Trimal and Mafft/Trimal Alignment

- Description:  

- Assumptions: 

- Limitations: 

_Code:_

```

#mafft-aligned Mr.Bayes tree:

#Downloaded MrBayes into the "data" folder. Using terminal, entered the "data" folder. Created a text file called "mbblock.txt" with the following commands pasted inside:
begin mrbayes;
 set autoclose=yes;
 prset brlenspr=unconstrained:exp(10.0);
 prset shapepr=exp(1.0);
 prset tratiopr=beta(1.0,1.0);
 prset statefreqpr=dirichlet(1.0,1.0,1.0,1.0);
 lset nst=2 rates=gamma ngammacat=4;
 mcmcp ngen=10000000 samplefreq=10 printfreq=100 nruns=1 nchains=3 savebrlens=yes;
 outgroup Antiqua;
 mcmc;
 sumt;
end;

#From the "data" folder, entered the

cat pestis-pMT1-all-mafft-trimal.nex mbblock.txt > pestis-pMT1-all-mafft-trimal-mrbayes.nex

mb pestis-pMT1-all-mafft-trimal-mrbayes.nex

```

```
#clustalw-aligned Mr.Bayes tree:

cat pestis-pMT1-all-clustalw-trimal.nex mbblock.txt > pestis-pMT1-all-clustalw-trimal-mrbayes.nex

mb pestis-pMT1-all-clustalw-trimal-mrbayes.nex

```

## Step 8: Coalescent Method Per each Parsimony and Distance Tree

**Completed- Astral per each Clustalw/Trimal/IQ-Tree and Mafft/Trimal/IQ-Tree Alignment

- Description:  

- Assumptions: 

- Limitations: 

_Code:_

```

#mafft-aligned IQ-Tree based Coalescent (Astral) tree:

#Installed Astral into "data" folder. Using terminal, went into the "Astral"" folder. Created a tre file containing mafft-aligned IQ-tree data in Newick format named "pestis-pMT1-all-mafft-trimal-iqtree.tre". Ran the following commands to get an output file called "pestis-pMT1-all-mafft-trimal-iqtree-astral.tre"

java -jar astral.5.7.8.jar -i pestis-pMT1-all-mafft-trimal-iqtree.tre -o pestis-pMT1-all-mafft-trimal-iqtree-astral.tre

#Copied the Newick format data from the "pestis-pMT1-all-mafft-trimal-iqtree-astral.tre" output file and ran the following commands in R to get a tree that was saved in desktop under "pMT-mafft-trimal-iqtree-astral.pdf"

library(ape)

library(adegenet)

library(phangorn)

mytree <- read.tree(text='(CP000309.1,(((CP010248.1,CP009714.1)0.67:0.2876820724517809,(NZ_CP045637.1,(NZ_CP016275.1,NC_009596.1)0.67:0.2876820724517809)0.67:0.2876820724517809)0.67:0.2876820724517809,(NC_014022.1,(CP000306.1,(NC_017155.1,((AE017045.1,NC_017266.1)0.67:0.2876820724517809,(CP045159.1,NC_003134.1)0.67:0.2876820724517809)0.67:0.2876820724517809)0.67:0.2876820724517809)0.67:0.2876820724517809)0.67:0.2876820724517809):0.0);')

plot(mytree)

nodelabels()

tre2 = root(mytree,outgroup="CP000309.1")

plot(tre2, cex = 0.6)
title("pMT mafft-aligned iqtree-based coalescent tree")

```

```
#clustalw-aligned IQ-Tree based Coalescent (Astral) tree:

#Astral was already downloaded into "data" folder. Using terminal, went into the "Astral"" folder. Created a tre file containing clustalw-aligned IQ-tree data in Newick format named "pestis-pMT1-all-clustalw-trimal-iqtree.tre". Ran the following commands to get an output file called "pestis-pMT1-all-clustalw-trimal-iqtree-astral.tre"

java -jar astral.5.7.8.jar -i pestis-pMT1-all-clustalw-trimal-iqtree.tre -o pestis-pMT1-all-clustalw-trimal-iqtree-astral.tre

#Copied the Newick format data from the "pestis-pMT1-all-clustalw-trimal-iqtree-astral.tre" output file and ran the following commands in R to get a tree that was saved in desktop under "pMT-clustalw-trimal-iqtree-astral.pdf"

library(ape)

library(adegenet)

library(phangorn)

mytree <- read.tree(text='(AE017045.1,(NC_017266.1,((NC_003134.1,CP045159.1)0.67:0.2876820724517809,((CP010248.1,CP009714.1)0.67:0.2876820724517809,((NZ_CP045637.1,(NZ_CP016275.1,NC_009596.1)0.67:0.2876820724517809)0.67:0.2876820724517809,((CP000306.1,NC_017155.1)0.67:0.2876820724517809,(NC_014022.1,CP000309.1)0.67:0.2876820724517809)0.67:0.2876820724517809)0.67:0.2876820724517809)0.67:0.2876820724517809)0.67:0.2876820724517809):0.0);')

plot(mytree)

nodelabels()

tre2 = root(mytree,outgroup="CP000309.1")

plot(tre2, cex = 0.6)
title("pMT clustalw-aligned iqtree-based coalescent tree")

```